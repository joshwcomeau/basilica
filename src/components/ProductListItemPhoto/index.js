/* eslint-disable jsx-a11y/no-static-element-interactions */
// eslint-disable-next-line no-unused-vars
import React, { Component, PropTypes } from 'react';
import classNames from 'classnames';

import './index.scss';


const ProductListItemPhoto = ({ imageUrl, altText, onClick }) => {
  const classes = classNames(['product-list-item-photo']);

  return (
    <a className={classes} onClick={onClick}>
      <img src={imageUrl} alt={altText} />
    </a>
  );
};

ProductListItemPhoto.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  altText: PropTypes.string,
  onClick: PropTypes.func,
};

ProductListItemPhoto.defaultProps = {

};

export default ProductListItemPhoto;
