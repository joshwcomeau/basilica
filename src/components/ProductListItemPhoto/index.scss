@import '../../variables';

.product-list-item-photo {


  @media screen and (max-width: $mq-tablet) {

  }

  @media screen and (max-width: $mq-mobile) {

  }
}
