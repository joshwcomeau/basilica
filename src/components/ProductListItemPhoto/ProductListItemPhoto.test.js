/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import ProductListItemPhoto from './index';

describe('ProductListItemPhoto', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<ProductListItemPhoto />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
