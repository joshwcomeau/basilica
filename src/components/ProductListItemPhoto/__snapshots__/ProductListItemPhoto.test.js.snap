exports[`ProductListItemPhoto renders correctly 1`] = `
<div
  className="product-list-item-photo">
  Your Component Here :)
</div>
`;
